includes/tracks/events/class-wc-admin-setup-wizard-tracking.php

```diff
@@ -19,9 +19,47 @@ public static function init() {
 			return;
 		}
 
+		add_action( 'wc_setup_footer', array( __CLASS__, 'add_footer_scripts' ) );
+		add_action( 'admin_init', array( __CLASS__, 'track_marketing_signup' ), 1 );
 		self::add_step_save_events();
 	}
 
+	/**
+	 * Get the name of the current step.
+	 *
+	 * @return string
+	 */
+	public static function get_current_step() {
+		return isset( $_GET['step'] ) ? sanitize_key( $_GET['step'] ) : ''; // phpcs:ignore WordPress.Security.NonceVerification.NoNonceVerification
+	}
+
+	/**
+	 * Add footer scripts to OBW since it does not contain hooks for
+	 * wp_footer to allow the default methods of enqueuing scripts.
+	 */
+	public static function add_footer_scripts() {
+		echo '<script type="text/javascript" src="https://stats.wp.com/w.js?ver=' . gmdate( 'YW' ) . '"></script>'; // @codingStandardsIgnoreLine
+		wc_print_js();
+	}
+
+	/**
+	 * Track the marketing form on submit.
+	 */
+	public static function track_marketing_signup() {
+		if ( 'next_steps' !== self::get_current_step() ) {
+			return;
+		}
+
+		wc_enqueue_js(
+			"
+			var form = $( '.newsletter-form-email' ).closest( 'form' );
+			$( document ).on( 'submit', form, function() {
+				window.wcTracks.recordEvent( 'obw_marketing_signup' );
```

Comment:
This function fires for me locally without error, but even running through the console it doesn't seem to show up in tracking history.  Is this being used incorrectly or is there something else happening here?

Reply:
>it doesn't seem to show up in tracking history

Are you referring to Tracks tools? If you are looking at the live page make sure you check "Filter for only rejected events". Events are rejected because the `wcadmin` prefix hasn't been whitelisted yet.

Alternatively, you can look at the network requests for `t.gif` and see the payload.

Reply:
I see the network requests, but don't see it under the live view (including rejected events).  But sometimes I notice these don't show up for a long time under live view.

Reply:
Sometimes it can take 10 minutes